src/App.xaml.cs

```diff
@@ -185,14 +184,14 @@ public void ShowMainWindow()
     {
         _dispatcherQueue.TryEnqueue(() =>
         {
-            var hWnd = WinRT.Interop.WindowNative.GetWindowHandle(MainWindow);
-            if (PInvoke.IsIconic(new HWND(hWnd)) && MainWindow.AppWindow.Presenter is OverlappedPresenter overlappedPresenter)
+            var hWnd = new HWND(WinRT.Interop.WindowNative.GetWindowHandle(MainWindow));
+            if (PInvoke.IsIconic(hWnd))
             {
-                overlappedPresenter.Restore(true);

```

Comment:
Good catch! I wonder if this is a bug in the WindowsAppSDK. Settings this to `Restore(false)` (without window activation) or calling `Restore()` and then bringing the window to the foreground works as expected.

```cs
var hWnd = new HWND(WinRT.Interop.WindowNative.GetWindowHandle(MainWindow));
if (PInvoke.IsIconic(hWnd) && MainWindow.AppWindow.Presenter is OverlappedPresenter overlappedPresenter)
{
    overlappedPresenter.Restore();
}

PInvoke.SetForegroundWindow(hWnd);
```

Calling `Restore(true)` does not restore a maximized screen correctly.

I created a bug in the corresponding repository to better understand this behavior:
- https://github.com/microsoft/WindowsAppSDK/issues/4570

Reply:
This seems to be caused by the window activation.
Let me know if `overlappedPresenter.Restore();` is preferred: I have used `PInvoke` since it's also used in case DevHome isn't minimized.